lib/livebook/runtime/evaluator.ex

```diff
@@ -454,6 +455,32 @@ defmodule Livebook.Runtime.Evaluator do
   defp code_error?(%CompileError{}), do: true
   defp code_error?(_error), do: false
 
+  defp reorder_binding(binding, prev_binding) do
+    # We keep the order of existing binding entries and move
+    # the new ones to the beginning, ordered alphabetically
+
+    binding_map = Map.new(binding)
+
+    prev_idx =
+      prev_binding
+      |> Enum.filter(fn {key, prev_val} ->
+        val = binding_map[key]
+        same?(val, prev_val)
+      end)
+      |> Enum.with_index()
+      |> Map.new(fn {{key, _val}, idx} -> {key, idx} end)
+
+    Enum.sort_by(binding, fn {key, _val} -> {prev_idx[key] || -1, key} end)
```

Comment:
I think there is likely a more efficient implementation by doing something like this: `remove_keys_in_old(new) ++ old`. WDYT?

Reply:
In this case, we wouldn't worry about rebinding, but I would say it is fine. If you want to keep rebinding, then you need to do a pass in `old` too.

Reply:
Good call, let me know if the new version makes sense :)